File: src/797. All Paths From Source to Target.js
/**
 * @param {number[][]} graph
 * @return {number[][]}
 */

export function allPathsSourceTarget(graph) {
  const result = [],
    path = [0],
    n = graph.length
  dfs(0)
  return result
  function dfs(node) {
    if (node === n - 1) {
      result.push([...path])
      return
    }
    for (let next of graph[node]) {
      path.push(next)
      dfs(next)
      path.pop()
    }
  }
}

File: src/797. All Paths From Source to Target.cpp
class Solution {
private:
    void dfs(vector<vector<int>>& graph, vector<vector<int>>& paths, vector<int> path, int index) {
        path.push_back(index);
        if (index == graph.size() - 1) {
            paths.push_back(path);
            return;
        }
        for (int nextIndex : graph[index]) {
            dfs(graph, paths, path, nextIndex);
        }
        path.pop_back();
    }
public:
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        vector<vector<int>> paths;
        vector<int> path;
        dfs(graph, paths, path, 0);
        return paths;
    }
};